vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -276,20 +270,13 @@ class OnboardingViewModel @AssistedInject constructor(
     private fun handleUserAcceptCertificate(action: OnboardingAction.UserAcceptCertificate) {
         // It happens when we get the login flow, or during direct authentication.
         // So alter the homeserver config and retrieve again the login flow
-        when (val finalLastAction = lastAction) {
-            is OnboardingAction.HomeServerChange.SelectHomeServer -> {
-                currentHomeServerConnectionConfig
-                        ?.let { it.copy(allowedFingerprints = it.allowedFingerprints + action.fingerprint) }
-                        ?.let { startAuthenticationFlow(finalLastAction, it, serverTypeOverride = null) }
-            }
+        when (action.retryAction) {
+            is OnboardingAction.HomeServerChange -> handleHomeserverChange(action.retryAction, fingerprint = action.fingerprint)
```

Comment:
`is OnboardingAction.HomeServerChange.SelectHomeServer` to `is OnboardingAction.HomeServerChange` is the fix, the other changes are refactors to enable unit testing the change